front/app/hooks/useEvents.ts

```diff
@@ -1,44 +1,68 @@
 import { useState, useEffect } from 'react';
 import { isNilOrError } from 'utils/helperUtils';
-import {
-  IEventData,
-  eventsStream,
-  IProjectsStreamParams,
-} from 'services/events';
-
-export default function useEvents(
-  projectIds?: string[],
-  futureOnly?: boolean,
-  pastOnly?: boolean
-) {
+import { getPageNumberFromUrl } from 'utils/paginationUtils';
+import { IEventData, eventsStream, IEventsStreamParams } from 'services/events';
+
+interface InputParameters {
+  projectIds?: string[];
+  futureOnly?: boolean;
+  pastOnly?: boolean;
+  pageNumber?: number;
+  pageSize?: number;
+}
+
+export default function useEvents({
+  projectIds,
+  futureOnly,
+  pastOnly,
+  pageNumber,
+  pageSize,
+}: InputParameters) {
   const [events, setEvents] = useState<IEventData[] | undefined | null | Error>(
     undefined
   );
+  const [lastPageNumber, setLastPageNumber] = useState(1);
```

Comment:
I would keep naming consistent with other resource hooks/components. So lastPageNumber should become lastPage. (In fact, I also prefer the more descriptive name pageNumber, but I think consistency is more important, so you could change them all in a different PR later.)